common/scala/src/main/scala/whisk/common/RandomString.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.common
+
+import java.security.SecureRandom
+
+import org.apache.commons.codec.binary.Hex
+
+object RandomString {
+  private val generator = new ThreadLocal[SecureRandom] {
```

Comment:
May be use [ThreadLocalRandom][1]. As we set `java.security.egd=file:/dev/./urandom` using normal Random may be just fine

[1]:  https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ThreadLocalRandom.html

Reply:
AFAIK `ThreadLocalRandom` uses a pseudo-random generator vs. `SecureRandom` actually uses `java.security.egd=file:/dev/./urandom`

Reply:
Checked again ... ThreadLocalRandom uses pseudo-random. So makes sense to implement a custom ThreadLocal impl